### PR TITLE
[BACKLOG-4040] - Review Component - List

### PR DESCRIPTION
**Pull request by diogofscmariano**
- Included Dev documentation in the ListBuilder

### CODE REVIEW

File: package-res/resources/web/prompting/builders/ListBuilder.js [hunk 1]
```diff
@@ -15,8 +15,45 @@
...
+ *   var listBuilder = new ListBuilder();
+ *
+ *   var listComponent = listBuilder.build(args);
+ * </code></pre>
+ *
+ * where 'args' is an object that contains the prompt panel and the parameters
+ * necessary for the component.
+ * The property multiSelect defines what is the component created.
+ * If multiSelect equals <code>true</code>, a {@link http://localhost:8080/pentaho/api/repos/:public:plugin-samples:pentaho-cdf:pentaho-cdf-require:30-documentation:30-component_reference:10-core:16-SelectComponent:select_component.xcdf/generatedContent SelectComponent} is created.
+ * If multiSelect equals <code>false</code>, a {@link http://localhost:8080/pentaho/api/repos/:public:plugin-samples:pentaho-cdf:pentaho-cdf-require:30-documentation:30-component_reference:10-core:19-SelectMultiComponent:select_multi_component.xcdf/generatedContent MultiSelectComponent} is created.
```
Comment (rfellows) [inline package-res/resources/web/prompting/builders/ListBuilder.js]: i think your links are opposite. if multi-select is TRUE, the SelectMultiComponent is used, not false.

File: package-res/resources/web/prompting/builders/ListBuilder.js [hunk 2]
```diff
@@ -15,8 +15,45 @@
...
+ *
+ * <p>
+ *   Note: the CDF documentation points to the Dashboard located on the Pentaho BI Server
+ * </p>
+ *
+ * @name ListBuilder
+ * @class
+ * @extends ParameterWidgetBuilderBase
```
Comment (annapopova1) [inline package-res/resources/web/prompting/builders/ListBuilder.js]: incorrect parent class name, used ValueBasedParameterWidgetBuilder for extending

**Comment by diogofscmariano** [comment]
@plagoa  @annapopova1 @rfellows @krivera-pentaho @scottyaslan @pamval please review, pr updated

**Comment by annapopova1** [comment]
@diogofscmariano Should we add js doc to "build" method or it's not necessary?

**Comment by diogofscmariano** [comment]
@annapopova1 yes, it probably does. I will update the pr and also add documentation for the superclasses

**Comment by diogofscmariano** [comment]
@plagoa @annapopova1 @rfellows @krivera-pentaho @scottyaslan @pamval please review, pr updated

File: package-res/resources/web/prompting/builders/ParameterWidgetBuilderBase.js [hunk 3]
```diff
@@ -16,14 +16,41 @@
...
+    /**
+     * Assigns to the widget to be built the main cdf properties needed to build a component
+     *
+     * @name ParameterWidgetBuilderBase#build
+     * @method
+     *
+     * @param {Object} args - The object with the properties to build the component
+     * @param {PromptPanel} args.promptPanel - The PromptPanel instance used to get the parameter name, the component name and the html object, where the component will be attached
+     * @param {Parameter} args.param - The parameter with the properties needed to build the component
+     * @returns {{promptType: string, executeAtStart: boolean, param: (args.param|{values, attributes}|{values, attributes, name}|{values, attributes, multiSelect}|{values}|{values, attributes, hasSelection}), name: *, htmlObject: *, type: undefined, parameter: *, postExecution: Function}}
```
Comment (annapopova1) [inline package-res/resources/web/prompting/builders/ParameterWidgetBuilderBase.js]: Is returns statement correct? In generated documentation it's represented as:
Returns:
Type
Object

**Comment by diogofscmariano** [comment]
@plagoa @annapopova1 @rfellows @krivera-pentaho @scottyaslan @pamval please review, pr updated

**Comment by annapopova1** [comment]
Looks good for me.

**Pull request merged**